include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -371,9 +611,188 @@ class alignment_file_format_sam
     //!\brief The format version string.
     static constexpr char format_version[4] = "1.6";
 
+    char buffer[100] = "";
+
     //!\brief A variable that tracks whether the header_ptr as been written or not.
     bool written_header{false};
 
+    template <typename stream_view_type, typename target_type>
+        requires detail::decays_to_ignore_v<target_type>
+    inline void read_field(stream_view_type && stream_view, target_type target)
+    {
+        detail::consume(stream_view);
+    }
+
+    template <typename stream_view_type, std::ranges::ForwardRange target_range_type>
+        requires requires (std::string s) { { view::char_to<value_type_t<target_range_type>>(s) }; }
```

Comment:
why this requirement?

Reply:
It works for all alphabets so it would be easier to just require that? Then again this is already enforced by our expensive checks on file-level.

Reply:
> Then again this is already enforced by our expensive checks on file-level.

Hm yes, I will remove the requires.